aicsimageio/readers/reader.py

```diff
@@ -247,6 +251,37 @@ def _read_immediate(self) -> xr.DataArray:
         """
         pass
 
+    def _get_stitched_dask_mosaic(self) -> xr.DataArray:
+        """
+        Stitch all mosaic tiles back together and return as a single xr.DataArray with
+        a delayed dask array for backing data.
+
+        Returns
+        -------
+        mosaic: xr.DataArray
+            The fully stitched together image. Contains all the dimensions of the image
+            with the YX expanded to the full mosaic.
+
+        Notes
+        -----
+        Implementers can determine how to chunk the array.
+        Most common is to chunk by tile.
+        """
+        pass
+
+    def _get_stitched_mosaic(self) -> xr.DataArray:
+        """
+        Stitch all mosaic tiles back together and return as a single xr.DataArray with
+        an in-memory numpy array for backing data.
+
+        Returns
+        -------
+        mosaic: np.ndarray
+            The fully stitched together image. Contains all the dimensions of the image
+            with the YX expanded to the full mosaic.
+        """
+        pass
```

Comment:
does pass return None or a default constructed return type?  If None, should the return value be Optional[xr.DataArray] ?  Idk how that works

Reply:
It returns `None`.

I am actually tempted to make this raise an `NotImplementedError` because this is an "optional" function (it doesn't have the `abstractmethod` decorator). This would mean that in the case someone tries to use the mosaic data on a reader that literally doesn't implement it, it raises error instead of just returning None. But opinions welcome.

Reply:
I was wondering if mosaic should just return full image in the case where it's not tiled.  But my question here was whether if pass returns None, then should the return type be Optional?  Or does mypy special-case it

Reply:
Hmmmmm I think this must be a mypy special case?